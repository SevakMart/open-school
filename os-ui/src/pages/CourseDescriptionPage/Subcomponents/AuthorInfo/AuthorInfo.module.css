.authorInformationContainer {
  margin-top: 56px;
}
.authorInformationContainer h2 {
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 600;
  font-size: 20px;
  line-height: 24px;
  color: #333941;
  margin: 0 0 16px;
}
.authorInformationContainer .authorInfo {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: none;
}
.authorInformationContainer .authorInfo .mentorProfileImage {
  border-radius: 50%;
  display: inline-block;
  width: 32px;
  height: 32px;
  border-color: black;
  margin-right: 16px;
}
.authorInformationContainer .authorInfo p {
  display: inline;
  font-family: "Montserrat";
  font-style: normal;
  font-weight: 500;
  font-size: 16;
  line-height: 24px;
  color: #5E617B;
  margin-right: 52px;
}
.authorInformationContainer .authorInfo a {
  cursor: pointer;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}/*# sourceMappingURL=AuthorInfo.module.css.map */