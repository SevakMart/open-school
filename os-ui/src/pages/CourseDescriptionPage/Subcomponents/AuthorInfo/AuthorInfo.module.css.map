{"version":3,"sources":["AuthorInfo.module.scss","AuthorInfo.module.css","../../../../styles/Mixins/FontMixin.scss","../../../../styles/Configs/Fonts.scss","../../../../styles/Configs/Colors.scss","../../../../styles/Mixins/FlexMixins.scss"],"names":[],"mappings":"AAKA;EACI,gBAAA;ACJJ;ADKI;EENF,yBCDY;EDEZ,kBFMuC;EELvC,gBFK8C;EEJ9C,eFIkD;EEHlD,iBFGuD;EACjD,cIJW;EJKX,gBAAA;ACCR;ADCI;EKXF,aAAA;EACA,mBLWyB;EKVzB,mBLU6B;EKT7B,qBLSoC;ACItC;ADHQ;EACI,kBAAA;EACA,qBAAA;EACA,WAAA;EACA,YAAA;EACA,mBAAA;EACA,kBAAA;ACKZ;ADHQ;EACI,eAAA;EEtBV,yBCDY;EDEZ,kBFsB2C;EErB3C,gBFqBkD;EEpBlD,aFoBsD;EEnBtD,iBFmByD;EAC/C,cIvBM;EJwBN,kBAAA;ACSZ;ADPQ;EACG,eAAA;EK5BT,aAAA;EACA,sBL4B4B;EK3B5B,mBL2BmC;EK1BnC,uBL0B0C;ACY5C","file":"AuthorInfo.module.css"}