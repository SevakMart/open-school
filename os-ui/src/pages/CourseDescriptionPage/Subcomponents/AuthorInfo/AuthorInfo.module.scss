@import "../../../../styles/Mixins/FlexMixins.scss";
@import "../../../../styles/Mixins/FontMixin.scss";
@import "../../../../styles/Configs/Colors.scss";
@import "../../../../styles/Configs/Fonts.scss";

.authorInformationContainer {
    margin-top:56px;
    h2 {
        @include fontStyle($primary-font,normal,600,20px,24px);
        color:$black-text-color;
        margin:0 0 16px;
    }
    .authorInfo {
        @include flexStyle(row,center,none);
        .mentorProfileImage {
            border-radius: 50%;
            display:inline-block;
            width:32px;
            height:32px;
            border-color: black;
            margin-right: 16px;
        }
        p {
            display:inline;
            @include fontStyle($primary-font,normal,500,16,20px);
            color:$main-text-color;
        }
        a {
           cursor:pointer;
           @include flexStyle(column,center,center);
        }
		.iconContent {
			height:36px;
			width:20%;
			margin-left:30px;
			@include flexStyle(row,center,space-evenly);
			.icon {
				display:inline-block;
				width:30px;
				height:30px;
				cursor: pointer;
			}
			
		}
    }
}