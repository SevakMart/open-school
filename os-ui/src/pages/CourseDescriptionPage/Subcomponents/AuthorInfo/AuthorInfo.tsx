import { useTranslation } from 'react-i18next';
import styles from './AuthorInfo.module.scss';
import LinkedinIcon1 from '../../../../icons/LinkedinIcon/Linkedin1';
import EmailIcon1 from '../../../../icons/EmailIcon/Email1';

const AuthorInfo = ({ mentorName, mentorSurname, mentorLinkedIn }:{mentorName:string, mentorSurname:string, mentorLinkedIn:string}) => {
  const { t } = useTranslation();
  const {
    authorInformationContainer, authorInfo, mentorProfileImage, icon, iconContent,
  } = styles;

  const handleLinkedinClick = () => {
    window.location.href = 'https://www.linkedin.com/';
  };

  return (
    <div className={authorInformationContainer}>
      <h2>{t('string.courseDescriptionPage.title.authorInfo')}</h2>
      <div className={authorInfo}>
        <img className={mentorProfileImage} src="https://reactjs.org/logo-og.png" alt={`${mentorName} ${mentorSurname}`} />
        <p>{`${mentorName} ${mentorSurname}`}</p>
        <div className={iconContent}>
          <div className={icon} onClick={handleLinkedinClick}><LinkedinIcon1 href={mentorLinkedIn} /></div>
          <div className={icon}><EmailIcon1 title="example@example.com" /></div>
        </div>
      </div>
    </div>
  );
};
export default AuthorInfo;
