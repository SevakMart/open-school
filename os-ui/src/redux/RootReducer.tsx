import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
