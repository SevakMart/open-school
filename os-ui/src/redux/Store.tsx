import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
