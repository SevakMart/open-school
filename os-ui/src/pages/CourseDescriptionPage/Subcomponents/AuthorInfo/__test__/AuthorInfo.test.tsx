import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { store } from '../../../../../redux/Store';
import AuthorInfo from '../AuthorInfo';

const mockUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom') as any,
  useNavigate: () => mockUseNavigate,
}));
jest.mock('react-i18next', () => ({
  ...jest.requireActual('react-i18next'),
  useTranslation: () => ({
    t: (message:string) => {
      if (message === 'error') return 'Something went wrong please refresh the page';
    },
  }),
}));

describe('Create test case to AuthorInfo component', () => {
  test('Create a snapshot test', async () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <AuthorInfo
            mentorName="John"
            mentorSurname="Smith"
            mentorLinkedIn="https://www.linkedin.com/in/john-smith/"
          />
        </Router>
      </Provider>,
    );
    expect(asFragment).toMatchSnapshot();
  });
});
