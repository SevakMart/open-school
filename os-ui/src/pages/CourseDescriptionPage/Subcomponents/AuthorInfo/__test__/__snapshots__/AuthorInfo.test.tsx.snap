// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test case to AuthorInfo component Create a snapshot test 1`] = `[Function]`;
